### PR TITLE
cEP-13.md: cobot enhancement and porting

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/coala/cEPs/issues/75

### CODE REVIEW

File: cEP-0009.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```
+> cobot ls bears [lang]*
+### lang1
+1. ...
+2. ...
+3. ...
+### lang2
+1. ...
+2. ...
+3. ...
```
Comment (meetmangukiya) [inline cEP-0009.md]: A link to all the bears

File: cEP-0009.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+Query the webservices with the parsed and retrieved code from the message and
+pass it along with the bears to be ran. Print the response that contains the
+diffs and results appropriately as a message.
+
+```
+> cobot run BearName setting1=value setting2=value ...
+..BearName2 setting1=value setting2=value ...
+```
+\#TODO confirm from hemangsk if passing settings will be possible in the
```
Comment (hemangsk) [inline cEP-0009.md]: Yes! 👍  By Phase 1

Comment (meetmangukiya) [reply cEP-0009.md]: Thanks! I didn't even have to ask :D

File: cEP-0009.md (hunk 1 above)
Comment (Udayan12167) [reply cEP-0009.md]: Can you describe how the output would look like and a brief overview of how the command will work on the backend. Like errbot will make a request to which api endpoint of webservices and so on

File: cEP-0009.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+|Status  |Proposed                               |
+|Type    |Process                                |
+
+Abstract
+--------
+
+This cEP describes the details of the process of porting
+[cobot](https://gitlab.com/coala/cobot) from hubot(coffeescript framework) to
+[errbot](http://errbot.io)(python framework) and about the new features that are
+to be added to cobot as a part of the
```
Comment (Udayan12167) [inline cEP-0009.md]: A sample code of a simple hello world command port from hubot to errbot should be mentioned here.

Comment (meetmangukiya) [reply cEP-0009.md]: Added a sample plugin example

File: cEP-0009.md (hunk 1 above)
Comment (meetmangukiya) [reply cEP-0009.md]: Added the endpoint that will be used and how will be the flow of the plugin

File: cEP-0009.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+1. ...
+2. ...
+3. ...
+See more at <link>
+```
+
+
+### stats(bears, languages, contribution)
+
+Query the webservices at appropriate end point according to the stat requested.
```
Comment (Udayan12167) [inline cEP-0009.md]: Even for this add endpoint.

Comment (meetmangukiya) [reply cEP-0009.md]: done :+1:

**Comment by jayvdb** [comment]
Instead of proscribing commands that interact with specific webservices, the webservices should be described using a webservice description language, or use REST, and then the bot generically maps those webservices into commands.

Then as new webservices are added, the bot only needs a restart or similar to load up the new commands.

**Comment by jayvdb** [comment]
The cEP should mention the obvious : We need a Python bot.

And then it should give a high level assessment of the most advanced python bots that exist, evaluating them, and explaining why errbot was selected.

**Comment by meetmangukiya** [comment]
> Instead of proscribing commands that interact with specific webservices, the webservices should be described using a webservice description language, or use REST, and then the bot generically maps those webservices into commands.

:+1: I will add this, I think I can fit this in the timeline somewhere

**Comment by Udayan12167** [comment]
ack https://github.com/coala/cEPs/pull/77/commits/9584f8ad223b467f106c8ab6127b85bcd2cebc67

**Comment by Udayan12167** [comment]
You need to rebase @meetmangukiya

**Comment by Udayan12167** [comment]
And get a review from @jayvdb

**Comment by Udayan12167** [comment]
ack https://github.com/coala/cEPs/pull/77/commits/9b274bcd2f4b4fcde625a7933fe1c35cf7818f69

**Comment by Udayan12167** [comment]
@rultor merge

**Pull request merged**